### DISCUSSION TITLE
How to set the widthxheight for webrtc publish page

### DISCUSSION THREAD

**Discussion by universesteve30** [ant-media/Ant-Media-Server#6621] [General]
Using the default webrtc publish page, I am constantly seeing Source WidthxHeight: 640 x 480 how to make the resolution larger because sending to Youtube is making the resolution very blurry but OBS is ok ?

**Comment by Mohit-3196** [comment]
Hi @universesteve30,
How are you?

Please check this discussion space where we have discussed the same feature https://github.com/orgs/ant-media/discussions/5337

Thank you

**Comment by SelimEmre** [comment]
Hi @universesteve30 and @Mohit-3196,

Thank you for the contribution, @Mohit-3196. I have some additional comments as well.

WebRTC dynamically selects resolution based on your network quality, available bandwidth, and camera capabilities. However, you can manually adjust these settings to better suit your needs. To do this, navigate to the following file:

```
/usr/local/antmedia/webapps/<App-Name>/samples/publish_webrtc.html
```

In this file, you can modify the `mediaConstraints` parameters as follows:

```javascript
var mediaConstraints = {
    audio: {
        noiseSuppression: true,
        echoCancellation: true
    },
    video: {width: 1920, height: 1080}
};
```

This example sets the video resolution to 1080p (1920x1080). You can adjust these values according to your requirements.

Additionally, you can change the `maxVideoBitrateKbps` variable to control the maximum video bitrate. The variable is located here:

```
https://github.com/ant-media/StreamApp/blob/master/src/main/webapp/samples/publish_webrtc.html#L203
```

While you can set this to 'unlimited' to allow the incoming bitrate to adapt based on network conditions, it's generally better to specify a reasonable limit. For example:

```javascript
var maxVideoBitrateKbps = 3000; // Set to 3 Mbps
```

Setting a specific value (like 2000 or 3000 Kbps) is considered good practice, as not all users may have sufficient bandwidth to handle very high bitrates. Choose a value that balances quality with accessibility for your target audience.

Remember that higher resolutions and bitrates will require more processing power and network bandwidth, so consider your users' typical devices and internet connections when configuring these settings.

Additionally, you can check the following pages for more details:
- https://webrtc.github.io/samples/src/content/getusermedia/resolution/
- https://test.8x8.vc/
- https://webrtchacks.com/video-constraints-2/

**Reply by rahul7827** [reply]
Thank you @SelimEmre. 

Happy to see your here. Your answer reminded me initial learning days in streaming. 

[Here it is](https://github.com/ant-media/Ant-Media-Server/issues/4063)
